### PR TITLE
chore(application-system): refactor and improve parental leave submit

### PR DESCRIPTION
**Pull request by baering**
## What

- Add and test util functions for edge cases when transforming application to parental leave to send to VMST
- Include a new data provider for family members
- Fix an issue in formatText

## Why

To improve how application is transformed into parental leave object that is sent to VMST

### New data provider for family members
image

image

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/application/template-api-modules/tsconfig.spec.json [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {
   "extends": "./tsconfig.json",
   "compilerOptions": {
+    "esModuleInterop": true,
```
Comment (baering) [inline libs/application/template-api-modules/tsconfig.spec.json:4]: To be able to `import get from 'lodash/get'` from unit test

File: libs/application/core/src/types/Form.ts [hunk 2]
```diff
@@ -5,7 +5,8 @@ import { MessageDescriptor } from 'react-intl'
 import { BoxProps } from '@island.is/island-ui/core'
 import { Application } from './Application'
 
-export type StaticText = (MessageDescriptor & { values?: object }) | string
+export type StaticTextObject = MessageDescriptor & { values?: object }
```
Comment (jeremybarbet) [inline libs/application/core/src/types/Form.ts:8]: Would be nicer to use `Record<string, unknown>` or probably somewhere a `RecordObject` exists somewhere iirc

File: libs/application/data-providers/src/providers/FamilyInformationProvider.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+        return Promise.resolve(response.data.nationalRegistryFamily)
+      })
+      .catch(() => {
+        return this.handleError()
+      })
+  }
+
+  handleError() {
+    return Promise.resolve([])
```
Comment (jeremybarbet) [inline libs/application/data-providers/src/providers/FamilyInformationProvider.ts:38]: No errors handlings?

File: libs/application/template-api-modules/src/lib/modules/templates/parental-leave/parental-leave.utils.ts [hunk 4]
```diff
@@ -77,6 +75,88 @@ const getEmployer = (
...
+    if (!spouse) {
+      throw new Error(
+        'transformApplicationToParentalLeaveDTO: Cannot find spouse. No family member with this relation.',
+      )
+    }
+
+    return spouse.nationalId
+  }
+
+  return otherParentId
```
Comment (jeremybarbet) [inline libs/application/template-api-modules/src/lib/modules/templates/parental-leave/parental-leave.utils.ts:113]: Is it fine to return an empty string if nothing can be found?

File: libs/application/templates/parental-leave/src/lib/messages.ts [hunk 5]
```diff
@@ -340,6 +340,16 @@ export const parentalLeaveFormMessages: MessageDir = {
       description:
         'In order to apply for this application we need your email and phone number',
     },
+    familyInformationTitle: {
+      id: 'pl.application:familyinformation.title',
+      defaultMessage: 'Upplýsingar um fjölskyldumeðlimi',
+      description: 'Your family information',
```
Comment (jeremybarbet) [inline libs/application/templates/parental-leave/src/lib/messages.ts:346]: Just for a reminder, the description is not used to pass the English translation, but more for the translations people to know what the message it is referring to. The description won't be used to populate the Contentful English field but just used to be displayed as the "description" section

**Review by jeremybarbet — COMMENTED** [review]
A few minor comments, overall really nice cleanup and PR 👍

File: libs/application/template-api-modules/src/lib/modules/templates/parental-leave/parental-leave.utils.ts (hunk 4 above)
Comment (baering) [reply libs/application/template-api-modules/src/lib/modules/templates/parental-leave/parental-leave.utils.ts:113]: Yes - I think so.

The otherParentId is required in the API payload but optional in the application since a primary parent is not required to enter the national registry id of the other parent.

File: libs/application/core/src/types/Form.ts (hunk 2 above)
Comment (baering) [reply libs/application/core/src/types/Form.ts:9]: Good catch. I gave it a go and it resulted in some weird ts errors in other places that I'll tackle some other time.

File: libs/application/data-providers/src/providers/FamilyInformationProvider.ts (hunk 3 above)
Comment (baering) [reply libs/application/data-providers/src/providers/FamilyInformationProvider.ts:38]: The `onProvideError` will get called with

```
{
  date: new Date(),
  reason: result,
  status: 'failure',
  data: result,
}
```

Where `result` is whatever handleErrors resolves to.

But I'll include logging when this happens.

**Review by madebynoam — APPROVED** [review]
Lgtm!

**Review by obmagnusson — APPROVED** [review]
LGTM
